### PR TITLE
Feature/#1284

### PR DESCRIPTION
**Pull request by nbon12**
Add API for getting AWS Account ID
Issue #1284 

- New table for "CloudAccounts" is created. The purpose of this table is to store cloud account IDs, and any other name-value pair.

- The columns are "Name", "Value", "Description"

- The first row in this table is the tuple: <"aws_account_id", "yourawsaccountidhere", "Cost Management's AWS Account ID">
- The only allowed operation is GET. To modify this value, Cost Management must log in to the postgres database directly.

Screen Shot 2019-10-25 at 3 18 49 PM

### CODE REVIEW

**Comment by nbon12** [comment]
Screen Shot 2019-10-22 at 9 44 47 AM

**Comment by blentz** [comment]
I'm not sure we need a whole new table for this. Or, perhaps we need to rework our existing Provider table to address this particular issue.

Within the role ARN that we already have is the account number.

The ARN format is this:
`arn:partition:service:region:account:resource`

So, when we get an ARN, like this:  `arn:aws:iam::123456789012:role/S3Access`, the account number is already embedded in the ARN. All we have to do is parse the ARN and extract the account number. 

We already have an ARN parsing class in our unit testing code (see: koku/masu/test/util/aws/test_common.py ). 

We can move that and use it for this purpose: to extract the Account number from the ARN we already have. Whether we store the account number separately in its own field, or just reparse ARNs whenever we need the account number is an open question we could talk about further.

File: koku/attribute/apps.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#
+# Copyright 2018 Red Hat, Inc.
```
Comment (dccurtis) [inline koku/attribute/apps.py]:
```suggestion
# Copyright 2019 Red Hat, Inc.
```

File: koku/attribute/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    operations = [
+        migrations.CreateModel(
+            name='Attribute',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(help_text='The name of the attribute', max_length=255)),
+                ('value', models.TextField(null=True)),
+                ('description', models.TextField(null=True)),
+                ('updated_timestamp', models.DateTimeField(auto_now=True, null=True)),
+                ('created_timestamp', models.DateTimeField(auto_now_add=True, null=True)),
```
Comment (dccurtis) [inline koku/attribute/migrations/0001_initial.py]: What's the reason for keeping track of two dates?  I can see maybe keeping one timestamp when the field is updated.

File: koku/attribute/models.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#
+# Copyright 2018 Red Hat, Inc.
```
Comment (dccurtis) [inline koku/attribute/models.py]:
```suggestion
# Copyright 2019 Red Hat, Inc.
```

File: koku/attribute/tests/test_model.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#
+# Copyright 2018 Red Hat, Inc.
```
Comment (dccurtis) [inline koku/attribute/tests/test_model.py]:
```suggestion
# Copyright 2019 Red Hat, Inc.
```

File: koku/attribute/urls.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#
+# Copyright 2018 Red Hat, Inc.
```
Comment (dccurtis) [inline koku/attribute/urls.py]:
```suggestion
# Copyright 2019 Red Hat, Inc.
```

File: koku/attribute/views.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+#
+# Copyright 2018 Red Hat, Inc.
```
Comment (dccurtis) [inline koku/attribute/views.py]:
```suggestion
# Copyright 2019 Red Hat, Inc.
```

File: koku/koku/settings.py [hunk 7]
```diff
@@ -86,6 +86,7 @@
     'reporting_common',
     'cost_models',
     'sources',
+    'attribute'
```
Comment (dccurtis) [inline koku/koku/settings.py]: I think this should probably live in the api app instead of a new Django app.

File: koku/attribute/urls.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""URLs for Attribute."""
+from rest_framework.routers import DefaultRouter
+from .views import AttributeViewSet
+from django.conf.urls import include, url
+
+ROUTER = DefaultRouter()
+ROUTER.register(r'attributes', AttributeViewSet, base_name='attributes')
```
Comment (dccurtis) [inline koku/attribute/urls.py]: the URL name `/attributes` isn't very descriptive as to what this API is for.

**Comment by nbon12** [comment]
Sure, name it whatever you want. I didn’t want to name it myself so I picked a generic name that can mean anything so that people can name things later, but if you want to name it in the codebase itself instead of giving ourselves the flexibility of naming things in the database, that’s fine too.

**Comment by nbon12** [comment]
> A few things.
> 
> The issue in 1284 is trying to address a specific "customer" requirement (even if the sources frontend team is our customer in this case) and I think we should try to focus specifically on that need. Unless you have good evidence for what other attributes we would be storing, I think we're over-complicating this issue by generalizing it.
> 
> Also, please move the API to the api app in django, not its own app.

@adberglund  What's the reason for including this in the api app instead of its own app?

**Comment by dccurtis** [comment]
@nbon12 I think you have to add the new endpoint here https://github.com/project-koku/koku/blob/09acee78d401a4656f555bab528eaf74bdabf449/koku/koku/middleware.py#L48

This shouldn't have any RBAC authorization since it's going to be called as an implementation detail to the frontend.

Hopefully this will fix the failing tests.

**Comment by adberglund** [comment]
Let's add the account ID we want to be returned by the API. Check out https://docs.djangoproject.com/en/2.2/topics/migrations/#data-migrations for how to use a Django data migration to do this.

**Comment by dlabrecq** [comment]
We would probably call this API lazily, when needed for the sources wizard. However, since the sources creation is being ported by the Catalog team, I don't know where else the koku-ui would use this? Regardless, the koku-i should cache all API calls for 30 min.

**Pull request merged**